x/oracle/keeper/reporter.go

```diff
@@ -14,6 +15,26 @@ func (k Keeper) IsReporter(ctx sdk.Context, val sdk.ValAddress, addr sdk.AccAddr
 	return ctx.KVStore(k.storeKey).Has(types.ReporterStoreKey(val, addr))
 }
 
+// AddReporter adds the reporter address to the list of reporters of the given validator.
+func (k Keeper) AddReporter(ctx sdk.Context, val sdk.ValAddress, addr sdk.AccAddress) error {
```

Comment:
Please update `keeper/msg_server.go` to call this function and delete that implementation.